Build error

This is the same error as #204 but I felt it might be good to revive it.

I'm on Unity 2021.3.44f1 and get the following 4 errors whenever I attempt to build:

```
Library/PackageCache/com.h8man.2d.navmeshplus@6d8836e05a/NavMeshComponents/Scripts/CollectTilemapSourcesCache2d.cs(30,68): error CS0426: The type name 'SyncTile' does not exist in the type 'Tilemap'
```
```
Error building Player because scripts had compiler errors
```
```
Build completed with a result of 'Failed' in 2 seconds (1564 ms)
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&) (at /home/bokken/build/output/unity/unity/Modules/IMGUI/GUIUtility.cs:189)
```
```
UnityEditor.BuildPlayerWindow+BuildMethodException: 2 errors
  at UnityEditor.BuildPlayerWindow+DefaultBuildMethods.BuildPlayer (UnityEditor.BuildPlayerOptions options) [0x002ce] in /home/bokken/build/output/unity/unity/Editor/Mono/BuildPlayerWindowBuildMethods.cs:194 
  at UnityEditor.BuildPlayerWindow.CallBuildMethods (System.Boolean askForBuildLocation, UnityEditor.BuildOptions defaultBuildOptions) [0x00080] in /home/bokken/build/output/unity/unity/Editor/Mono/BuildPlayerWindowBuildMethods.cs:95 
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&) (at /home/bokken/build/output/unity/unity/Modules/IMGUI/GUIUtility.cs:189)
```

--- a/NavMeshComponents/Scripts/CollectTilemapSourcesCache2d.cs
+++ b/NavMeshComponents/Scripts/CollectTilemapSourcesCache2d.cs
@@ -27,6 +27,7 @@ namespace NavMeshPlus.Extensions
             base.Awake();
         }
 
+#if UNITY_EDITOR || UNITY_2022_2_OR_NEWER && UNITY_EDITOR
         private void OnTilemapTileChanged(Tilemap tilemap, Tilemap.SyncTile[] syncTiles)
         {
             if (tilemap == _tilemap)
@@ -51,6 +52,8 @@ namespace NavMeshPlus.Extensions
                 }
             }
         }
+#endif
+
 
         public AsyncOperation UpdateNavMesh(NavMeshData data)
         {
@@ -75,13 +78,17 @@ namespace NavMeshPlus.Extensions
                     _lookup[position] = i;
                 }
             }
+            #if UNITY_EDITOR || UNITY_2022_2_OR_NEWER && UNITY_EDITOR
             Tilemap.tilemapTileChanged -= OnTilemapTileChanged;
             Tilemap.tilemapTileChanged += OnTilemapTileChanged;
+            #endif
         }
 
         protected override void OnDestroy()
         {
+            #if UNITY_EDITOR || UNITY_2022_2_OR_NEWER && UNITY_EDITOR
             Tilemap.tilemapTileChanged -= OnTilemapTileChanged;
+            #endif
             base.OnDestroy();
         }
     }

--- a/NavMeshComponents/Scripts/CollectTilemapSourcesCache2d.cs
+++ b/NavMeshComponents/Scripts/CollectTilemapSourcesCache2d.cs
@@ -27,7 +27,7 @@ namespace NavMeshPlus.Extensions
             base.Awake();
         }
 
-#if UNITY_EDITOR || UNITY_2022_2_OR_NEWER && UNITY_EDITOR
+#if UNITY_EDITOR || UNITY_2022_2_OR_NEWER
         private void OnTilemapTileChanged(Tilemap tilemap, Tilemap.SyncTile[] syncTiles)
         {
             if (tilemap == _tilemap)
@@ -78,7 +78,7 @@ namespace NavMeshPlus.Extensions
                     _lookup[position] = i;
                 }
             }
-            #if UNITY_EDITOR || UNITY_2022_2_OR_NEWER && UNITY_EDITOR
+            #if UNITY_EDITOR || UNITY_2022_2_OR_NEWER
             Tilemap.tilemapTileChanged -= OnTilemapTileChanged;
             Tilemap.tilemapTileChanged += OnTilemapTileChanged;
             #endif
@@ -86,7 +86,7 @@ namespace NavMeshPlus.Extensions
 
         protected override void OnDestroy()
         {
-            #if UNITY_EDITOR || UNITY_2022_2_OR_NEWER && UNITY_EDITOR
+            #if UNITY_EDITOR || UNITY_2022_2_OR_NEWER
             Tilemap.tilemapTileChanged -= OnTilemapTileChanged;
             #endif
             base.OnDestroy();

